core/domain/stats_services.py

```diff
@@ -1024,6 +1026,67 @@ def _get_calc_output(exploration_id, state_name, calculation_id):
         return None
 
 
+def validate_interaction_id_with_state_for_exploration(
```

Comment:
As mentioned above, this (or a variant of it) should go in exp_services. It has nothing to do with stats.

Reply:
Done